### PR TITLE
feature/implement-connection-screen

### PR DESCRIPTION
**Pull request by Czaroos**
- Added connect with github popup when logged user isn't already connected to github.
- It appears every time new session is created.
- Added removeCookie() to CookiesProvider

### CODE REVIEW

File: src/core/cookies/CookiesProvider.tsx [hunk 1]
```diff
@@ -25,6 +27,12 @@ const STATE: CookiesProvider.State = {
 const Context = createContext(STATE);
 
 class Provider extends React.Component<CookiesProvider.Props, typeof STATE> {
+  removeCookie: CookiesProvider.RemoveCookie = (name) => {
+    Cookies.remove(name);
+
```
Comment (ajronn) [inline src/core/cookies/CookiesProvider.tsx:32]: Remove space

**Review by ajronn — APPROVED** [review]

File: src/shared/components/github-connect/GithubConnect.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import React, { useCallback, useState, useEffect } from 'react';
+
+import { Modal, Button } from 'ui';
+
+import { useAuthProvider } from 'core/auth';
+import { useCookiesProvider } from 'core/cookies';
+
+import csx from './GithubConnect.scss';
+
+interface State {
```
Comment (polubis) [inline src/shared/components/github-connect/GithubConnect.tsx:10]: Please change this to simple flag and state to status and setStatus - object is not needed here

File: src/shared/components/github-connect/GithubConnect.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  success: boolean;
+}
+
+const STATE: State = {
+  success: false
+};
+
+const NAME = 'github-connection';
```
Comment (polubis) [inline src/shared/components/github-connect/GithubConnect.tsx:18]: Can be 1 lined with array syntax [NAME, NOT_NOW_VALUE] = [''. '']

File: src/shared/components/github-connect/GithubConnect.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+  const { user, logInViaGithub } = useAuthProvider();
+
+  const { cookies, setCookies, removeCookie } = useCookiesProvider();
+
+  useEffect(() => {
+    user && user.connectedWithGithub && setState({ success: true });
+  }, [user]);
+
+  const connect = useCallback(() => {
```
Comment (polubis) [inline src/shared/components/github-connect/GithubConnect.tsx:33]: handleConnect convention

File: src/shared/components/github-connect/GithubConnect.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    setCookies(NAME, user.id);
+    logInViaGithub();
+  }, [user]);
+
+  const clearCookies = useCallback(() => {
```
Comment (polubis) [inline src/shared/components/github-connect/GithubConnect.tsx:38]: handleClearCookies

File: src/shared/components/github-connect/GithubConnect.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    setState(STATE);
+    removeCookie(NAME);
+  }, []);
+
+  const notNow = useCallback(() => {
```
Comment (polubis) [inline src/shared/components/github-connect/GithubConnect.tsx:43]: handleNotNow

File: src/shared/components/github-connect/GithubConnect.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    setCookies(NAME, NOT_NOW_VALUE);
+  }, []);
+
+  const shouldShowScreen =
+    (user && !user.connectedWithGithub && cookies[NAME] !== NOT_NOW_VALUE) ||
+    (user && cookies[NAME] === user.id && success);
+
+  return (
+    shouldShowScreen && (
```
Comment (polubis) [inline src/shared/components/github-connect/GithubConnect.tsx:52]: This should be shouldShowScreen ? content : null;

File: src/shared/components/github-connect/GithubConnect.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  const clearCookies = useCallback(() => {
+    setState(STATE);
+    removeCookie(NAME);
+  }, []);
+
+  const notNow = useCallback(() => {
+    setCookies(NAME, NOT_NOW_VALUE);
+  }, []);
+
+  const shouldShowScreen =
```
Comment (polubis) [inline src/shared/components/github-connect/GithubConnect.tsx:47]: visible will be enough for this variable name

Comment (polubis) [reply src/shared/components/github-connect/GithubConnect.tsx:47]: can be simplified with user && ((!user.connectedWithGithub && cookies[NAME] !== NOT_NOW_VALUE) || cookies[NAME] === user.id && success) -

File: src/shared/components/github-connect/GithubConnect.tsx [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+interface State {
+  success: boolean;
+}
+
+const STATE: State = {
+  success: false
+};
+
+const NAME = 'github-connection';
+const NOT_NOW_VALUE = 'not now';
```
Comment (polubis) [inline src/shared/components/github-connect/GithubConnect.tsx:19]: Change value to '1' and name of this variable APPROVED

File: src/shared/components/github-connect/GithubConnect.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+          <>
+            <h3>
+              Connect your account to <span>Github</span>
+            </h3>
+            <p>
+              Our system has detected that you are not yet connected to <span>Github</span>. This is
+              not a problem if you want to use basic features.
+            </p>
+            <p>You can connect now or do it later.</p>
+            <div className={csx.buttons}>
```
Comment (polubis) [inline src/shared/components/github-connect/GithubConnect.tsx:64]: You can use footer and avoid not needed classNames - this also increases markdown readability

File: src/shared/components/github-connect/GithubConnect.tsx [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+  }, []);
+
+  const shouldShowScreen =
+    (user && !user.connectedWithGithub && cookies[NAME] !== NOT_NOW_VALUE) ||
+    (user && cookies[NAME] === user.id && success);
+
+  return (
+    shouldShowScreen && (
+      <Modal className={csx.githubConnect}>
+        {!success ? (
```
Comment (polubis) [inline src/shared/components/github-connect/GithubConnect.tsx:54]: Remove negation before success and swap contents

File: src/modules/main/Main.tsx [hunk 12]
```diff
@@ -70,6 +71,7 @@ const Main = ({ match }: RouteChildrenProps) => {
 
 export default (props: RouteChildrenProps) => (
   <NotificationsProvider>
+    <GithubConnect />
```
Comment (polubis) [inline src/modules/main/Main.tsx:74]: Add this inside Main just after Sidebar - using this in this place tells me GithubConnect needs Provider data to work

**Review by polubis — CHANGES_REQUESTED** [review]

**Review by i0skar — APPROVED** [review]

**Pull request merged**